Boxed icon component

chancovsky opened the pull request:
# Boxed icon component

Figma reference: https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=448%3A704

ajmnz commented on kibbeh/src/ui/BoxedIcon.tsx:
```diff
@@ -0,0 +1,18 @@
+export const BoxedIcon: React.FC<BoxedIconProps> = ({ icon, ...props }) => {
+  return (
+    <div
+      className="bg-primary-700 justify-center items-center rounded"
```
Set the background to `gray-600` on hover and use `w-6 h-6` instead of inline styles https://github.com/benawad/dogehouse/blob/24d03500c4b4a84228cb897e9e30b8ac3304664d/kibbeh/tailwind.config.js#L42

Also `cursor-pointer`.

benawad commented on kibbeh/tailwind.config.js:
```diff
@@ -27,6 +27,15 @@ module.exports = {
+        700: "var(--color-gray-700)",
+        800: "var(--color-gray-800)",
+        900: "var(--color-gray-900)",
+      },
```
use primary instead of gray

benawad commented on kibbeh/src/stories/BoxedIcon.story.tsx:
```diff
@@ -0,0 +1,58 @@
+      <div className="m-1">
+        <BoxedIcon
+          icon={
+            <svg width="14" height="18" viewBox="0 0 14 18" fill="none">
```
Use the React components inside `src/icons` for icons

ajmnz commented:
Why did you change the icon sizes? That affected yours and other already finished components. Each icon is ready to be imported as is, with its width and height already set up.

chancovsky commented:
I changed the Icon sizes to normalize them against `font-size`.  This will allow for more consistent icon sizing across the application instead of having icons sized for specific use cases.  It also allows for more flexibility when using the icons and have a consistent way manage their size.

ajmnz commented:
> I changed the Icon sizes to normalize them against `font-size`. This will allow for more consistent icon sizing across the application instead of having icons sized for specific use cases. It also allows for more flexibility when using the icons and have a consistent way manage their size.

~~Makes sense, but we will be doing that in the future to ensure every icon remains the size it was meant to be. Right now the priority is to have the components finished exactly like they are on Figma.~~

Let's do it, but revert that default `#FFF` fill from the svgs. Pure white only applies to the red button children.

benawad commented on kibbeh/src/icons/solid-rocket.svg:
```diff
@@ -0,0 +1,4 @@
+<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M36.0783 4.1874C36.0471 4.05459 35.9533 3.95303 35.8205 3.92178C31.5705 2.88271 21.7502 6.58584 16.4299 11.9061C15.3908 12.9452 14.5392 13.9686 13.8439 14.9608C12.2033 14.8124 10.5627 14.9374 9.16423 15.5468C5.21892 17.2811 4.07047 21.8124 3.75017 23.7577C3.67204 24.2187 4.05485 24.6171 4.51579 24.5702L10.8517 23.8749C10.8595 24.4843 10.8986 24.9686 10.9376 25.3046C10.9689 25.6327 11.117 ...
```
can you remove the raw svgs

chancovsky replied:
Do we want to remove all the raw svgs from the repo or just relocate them to an `assets` folder or something like that and just have the svg react components in `/icons`?

benawad commented on kibbeh/svgr.config.js:
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  typescript: true,
+  replaceAttrValues: { "#000": "currentColor" },
```
why did you change from `#fff` to `#000`?

chancovsky replied:
mistake, changed to `#fff`

benawad commented:
The icon size normalization, breaks some other components
![image](https://user-images.githubusercontent.com/7872329/110549795-3555eb00-80f8-11eb-8025-ce999f834114.png)
![image](https://user-images.githubusercontent.com/7872329/110549819-3f77e980-80f8-11eb-8d13-7ca6ef03ff7c.png)

The current icons are "draft" icons so they are going to get switched out in the future. It's probably best to scrap the normalization for now and add it in when the "prod" icons are in.

chancovsky commented:
> The icon size normalization, breaks some other components
> ![image](https://user-images.githubusercontent.com/7872329/110549795-3555eb00-80f8-11eb-8025-ce999f834114.png)
> ![image](https://user-images.githubusercontent.com/7872329/110549819-3f77e980-80f8-11eb-8d13-7ca6ef03ff7c.png)
> 
> The current icons are "draft" icons so they are going to get switched out in the future. It's probably best to scrap the normalization for now and add it in when the "prod" icons are in.

The ScheduledTime is an alignment issue outside of svgs that I fixed. And I adjusted the dogenitro svg sizing a bit to be inline with the designs.  I can revert this all back to the original svgs I you think we aren't ready to use normalized icons.

benawad commented:
yeah if you wouldn't mind reverting, I want to do the normalization at a later point

benawad commented:
Also is it possible to cleanup the changes so it just shows the stuff you changed.
